pilot/pilot/core.py

```diff
@@ -842,25 +842,13 @@ def _cache_safe(self, key, func, *arguments):
             finally:
                 self._cache[key] = 1
 
-    def _process_ros(self, c_ros):
-        # It could be a collection of ros commands.
-        ros_messages = [] if c_ros is None else c_ros if (isinstance(c_ros, tuple) or isinstance(c_ros, list)) else [c_ros]
-        for _msg in ros_messages:
-            if "pilot.driver.set" in _msg:
-                self._cache_safe("ros switch driver", lambda: self._driver.switch_ctl(_msg.get("pilot.driver.set")))
```

Comment:
Are you sure that this key `pilot.driver.set` isn't used anywhere else in the code?

please check the implementation of the original function https://github.com/MWLCDev/BYODR/blob/7a1a6f4a0a3f253b58c74732ecc9b49cfdc93e47/pilot/pilot/core.py#L748
 and see what data is in the `_msg` variable of `ros_message`

Reply:
`def _publish_diagnostics(self):
        pilot = self._get_pilot()
        if pilot:
            status = DiagnosticStatus(name='mode', message='not_available')
            status.values = [
                KeyValue(key='steer', value='{:+2.5f}'.format(pilot.get('steering'))),
                KeyValue(key='throttle', value='{:+2.5f}'.format(pilot.get('throttle')))
            ]
            driver_mode = pilot.get('driver')
            if driver_mode == 'driver_mode.teleop.direct':
                status.message = 'teleoperation'
            elif driver_mode == 'driver_mode.inference.dnn':
                status.message = 'autopilot'
                status.values += [
                    KeyValue(key='maximum_speed', value='{:+2.5f}'.format(pilot.get('cruise_speed'))),
                    KeyValue(key='desired_speed', value='{:+2.5f}'.format(pilot.get('desired_speed')))
                ]
            diagnostics = DiagnosticArray()
            diagnostics.header.stamp = self.get_clock().now().to_msg()
            diagnostics.status = [status]
            self._ros_publisher.publish(diagnostics)`